### PR TITLE
Add ability to configure where global npmrc lines are added

### PR DESCRIPTION
**Pull request by william2958**
## Summary
This MR adds the option to specify lines to be prepended and/or appended to the .npmrc file when it is synced via the Utilities.syncNpmrc function from the source to target folder. It also cleans up the syncNpmrc function parameter list by moving the parameters into an option object.

This is so that global npmrc lines can be added before subspace npmrc lines, allowing for subspace level configuration overrides. This MR also adds the global npmrc file to rush init.

### CODE REVIEW

File: libraries/rush-lib/src/utilities/npmrcUtilities.ts [hunk 1]
```diff
@@ -22,13 +22,25 @@ export interface ILogger {
...
 const _combinedNpmrcMap: Map<string, string> = new Map();
 
-function _trimNpmrcFile(sourceNpmrcPath: string, extraLines: string[] = []): string {
+function _trimNpmrcFile(
+  sourceNpmrcPath: string,
+  extraLines: {
+    lines: string[];
+    insertBefore: boolean;
+  } = { lines: [], insertBefore: false }
+): string {
```
Comment (octogonz) [inline libraries/rush-lib/src/utilities/npmrcUtilities.ts:31]: This function signature is getting messy.  Maybe we can do something like:

```ts
function _trimNpmrcFile(options: {
  sourceNpmrcPath: string,
  linesToPrepend?: string[],
  linesToAppend?: string[],
): string {
```

File: libraries/rush-lib/src/logic/base/BaseInstallManager.ts [hunk 2]
```diff
@@ -391,7 +391,10 @@ export abstract class BaseInstallManager {
       subspace.getSubspaceTempFolder(),
       undefined,
       undefined,
-      extraNpmrcLines
```
Comment (octogonz) [inline libraries/rush-lib/src/logic/base/BaseInstallManager.ts:394]:
```ts
      // Look for a global .npmrc-global file
      const globalNpmrcPath: string = `${this.rushConfiguration.commonRushConfigFolder}/.npmrc-global`;
```

@william2958 Let's update [rfc-4230-rush-subspaces.md](https://github.com/microsoft/rushstack/blob/main/common/docs/rfcs/rfc-4230-rush-subspaces.md) to document this file (and any other new config files introduced by the Subspaces feature).

Adding a `rush init` template would also be a good form of documentation, although if you think it's likely that the file path/semantics might change in the future, we can procrastinate that.

File: common/changes/@microsoft/rush/will-subspace-patch-8_2024-03-07-22-37.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Add configuration options to append or prepend extra lines to the npmrc when it is synced from the source to target directory.",
```
Comment (octogonz) [inline common/changes/@microsoft/rush/will-subspace-patch-8_2024-03-07-22-37.json:5]:
```suggestion
      "comment": "Fix an issue where, when the experimental subspace feature is enabled, the subspace's \".npmrc\" file did not take precedence over \".npmrc-global\".",
```

File: libraries/rush-lib/src/utilities/npmrcUtilities.ts (hunk 1 above)
Comment (william2958) [reply libraries/rush-lib/src/utilities/npmrcUtilities.ts]: Changed to use an option object

**Pull request merged**